From 9381f1943a9e2c5273e32a76fae36daf67ab7089 Mon Sep 17 00:00:00 2001
From: Miroslav Rezanina <mrezanin@redhat.com>
Date: Tue, 24 Oct 2017 05:33:03 +0200
Subject: [PATCH 26/27] target-i386: Enable clflushopt/clwb/pcommit
 instructions

RH-Author: Eduardo Habkost <ehabkost@redhat.com>
Message-id: <20171017203653.28578-2-ehabkost@redhat.com>
Patchwork-id: 77349
O-Subject: [RHEL-7.5 qemu-kvm PATCH 1/2] target-i386: Enable clflushopt/clwb/pcommit instructions
Bugzilla: 1501510
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Igor Mammedov <imammedo@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>

From: Xiao Guangrong <guangrong.xiao@linux.intel.com>

These instructions are used by NVDIMM drivers and the specification is
located at:
https://software.intel.com/sites/default/files/managed/0d/53/319433-022.pdf

There instructions are available on Skylake Server.

Signed-off-by: Xiao Guangrong <guangrong.xiao@linux.intel.com>
Reviewed-by: Richard Henderson <rth@twiddle.net>
Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
(cherry picked from commit f7fda280948a5e74aeb076ef346b991ecb173c56)
Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 target-i386/cpu.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/target-i386/cpu.c b/target-i386/cpu.c
index 914fc66..e739647 100644
--- a/target-i386/cpu.c
+++ b/target-i386/cpu.c
@@ -149,8 +149,8 @@ static const char *cpuid_7_0_ebx_feature_name[] = {
     "bmi2", "erms", "invpcid", "rtm",
     NULL, NULL, "mpx", NULL,
     "avx512f", "avx512dq", "rdseed", "adx",
-    "smap", "avx512ifma", NULL, NULL,
-    NULL, NULL, "avx512pf", "avx512er",
+    "smap", "avx512ifma", "pcommit", "clflushopt",
+    "clwb", NULL, "avx512pf", "avx512er",
     "avx512cd", "sha-ni", "avx512bw", "avx512vl",
 };
 
-- 
1.8.3.1

